Themis/Views/Assessment/AssessmentView.swift

```diff
@@ -276,8 +283,12 @@ struct AssessmentView: View {
             }
         }
         .task(priority: .high) {
+            if let submissionId, vm.submission == nil {
+                await vm.getSubmission(id: submissionId)
+            }
```

Comment:
I don't like that this moved from the `AssessmentSubmissionLoaderView` to here. The `AssessmentView` is pretty large as-is. Why couldn't we keep it there?

Reply:
As you already noticed below `CodeEditorViewModel` is a StateObject inside AssessmentViewModel now which means it is properly reinitialised every time a new submission is opened. This solves the bug that things such as inline feedbacks for example were not properly removed when assessing another submission.